archspec/cpu/microarchitecture.py

```diff
@@ -218,15 +229,9 @@ def satisfies_constraint(entry, version):
             min_version, max_version = entry["versions"].split(":")
 
             # Check version suffixes
-            min_version, min_suffix = version_components(min_version)
-            max_version, max_suffix = version_components(max_version)
-            version, suffix = version_components(version)
-
-            # If the suffixes are not all equal there's no match
-            if (suffix != min_suffix and min_version) or (
-                suffix != max_suffix and max_version  # pylint: disable=bad-continuation
-            ):
-                return False
+            min_version, _ = version_components(min_version)
+            max_version, _ = version_components(max_version)
+            version, _ = version_components(version)
 
             # Assume compiler versions fit into semver
             tuplify = lambda x: tuple(int(y) for y in x.split("."))  # noqa: E731,E501
```

Comment:
Just use a def and get rid of the noqa here.  Also, do not use any E501's!  Just shorten the line!

Reply:
Done. This was like that since 2c36f2a2a (the initial commit).